packages/core/src/request.ts

```diff
@@ -161,6 +163,36 @@ export class Request {
         this.headers = { ...headers };
         this.userData = { ...userData };
         this.handledAt = handledAt as unknown instanceof Date ? (handledAt as Date).toISOString() : handledAt!;
+
+        Object.defineProperties(this.userData, {
+            __crawlee: {
+                // eslint-disable-next-line no-underscore-dangle
+                value: this.userData.__crawlee,
+                enumerable: false,
+                writable: true,
+            },
```

Comment:
I think I imagined that the `__crawlee` property would not be there at all until we serialize the request with toJSON, but maybe this is useful for something? @B4nan

Reply:
Oh, I see you discussed it in the other thread. Well, as I said, I thought that we would only add the property when serializing with toJSON, but I guess I don't mind that it's there when it's not enumerable.

Reply:
We could move this to the setter, so we have the value only when needed.

Reply:
Thinking about it now, I'm not sure if setting `toJSON` on the `userData` object is correct. What if the user does the following. I'm pretty sure there will be code like that in our actors.

```js
request.userData = { foo: "bar" }
```

I think we need to add the toJSON function directly to the Request class. @barjin

Reply:
This is definitely true! The proposed solution however breaks [this line](https://github.com/apify/apify-storage-local-js/blob/6083615fa92987f880e19755fb7d9a3818355e4d/src/resource_clients/request_queue.ts#L234) in `apify-storage-local`, as the Request instance is being spread - the `toJSON` property gets spread as well and overshadows the new `id` field. 

Setter/getter pair for `userData` would not work either, as that would require custom `Request.toJSON()` for the set/get value to appear in the serialized data as well. 

Perhaps we should extend the Request interface and use the Request class more rigidly throughout the packages (no more spreads etc.)?

Reply:
> Setter/getter pair for userData would not work either, as that would require custom Request.toJSON() for the set/get value to appear in the serialized data as well.

Just mark it explicitly as `enumerable: true`, it will get serialized correctly.

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAJgQygmBeGBvAvjBFdgCeAsAFCiSwhqY54EmlkDyARgFYCmwUAdHJwBmASzCcACgCcQAB06SohABQgANDADkgkCA3qMZGEZgBzTlABcMJQEo0APnhIEvAG4IANqsPGI5q0qudqiOiMhunjSu3qTGMJxgAK4AtvIIrB6cVlCSiZwxWDZkZBQQIJm8HiAmKuoAUgDKzAByvNCSoibCgsogNkVMpCC82tToAIwATADMxeTgZRVVNWowjS1tOZ3dvf1AA